src/Psalm/Internal/Analyzer/Statements/Expression/SimpleTypeInferer.php

```diff
@@ -203,6 +203,61 @@ public static function infer(
             }
         }
 
+        if ($stmt instanceof PhpParser\Node\Expr\BitwiseNot) {
+            $stmt_expr_type = self::infer(
+                $codebase,
+                $nodes,
+                $stmt->expr,
+                $aliases,
+                $file_source,
+                $existing_class_constants,
+                $fq_classlike_name
+            );
+
+            if ($stmt_expr_type === null) {
+                return null;
+            }
+
+            $invalidTypes = clone $stmt_expr_type;
+            $invalidTypes->removeType('string');
+            $invalidTypes->removeType('int');
+            $invalidTypes->removeType('float');
```

Comment:
Is float actually a possible return type for the `~` operator?

I thought I actually added an issue about this, but apparently I only [considered it](https://github.com/vimeo/psalm/pull/7992#issuecomment-1133272149) then never did it (went ahead and added #8362). No need to worry about fixing that here though, just need to remove the `float` result type.

Reply:
Nevermind, I see what's going on, this is just checking if the operand is valid.